lib/galaxy/webapps/galaxy/controllers/async.py

```diff
@@ -121,7 +121,14 @@ def index(self, trans, tool_id=None, data_secret=None, **kwd):
                 galaxy_url  = trans.request.base + '/async/%s/%s/%s' % ( tool_id, data.id, key )
                 params.update( { 'GALAXY_URL' :galaxy_url } )
                 params.update( { 'data_id' :data.id } )
-                url  = tool.action + '?' + urllib.urlencode( params.flatten() )
+                # Use provided URL or fallback to tool action
+                url = URL or tool.action
+                # Does url already have query params?
+                if '?' in url:
+                    url_join_char = '&'
+                else:
+                    url_join_char = '?'
+                url = "%s%s%s" % ( url, url_join_char, urllib.urlencode( params.flatten() ) )
```

Comment:
I mean... I'd personally prefer proper url parsing via urllib2 (or similar) buuuut it works, so good enough for me.